### ISSUE TITLE
Multi-root workspace: single language server bootstrapped from shortest-path folder causes false import errors in other folders

### ISSUE THREAD

**Issue by suliatmayowaogungbe-glitch** [microsoft/vscode-pylint#872]
### Expected Behavior
In a multi-root VS Code workspace, each root folder's files are linted using that folder's own configured Python interpreter/environment, so packages installed (including editable, `pip install -e .`) in one folder's venv don't cause false import errors when linting a *different* folder that has its own separate venv.

### Actual Behavior
Pylint runs a single shared language server process for the whole workspace, bootstrapped from exactly one root folder — the one with the shortest filesystem path (`getProjectRoot()` in `vscode-common-python-lsp`'s `utilities.ts`). Every other root folder in the workspace is linted through that same process, using that one folder's interpreter. If a different root folder has its own separate venv with its own packages (including editable installs), files in that folder get a false `E0401: Unable to import '<pkg>'` error, because the process is running under the wrong interpreter entirely — independent of any `pylint.importStrategy` setting.

## Reproduction Steps

1. Create two separate folders, each with its own venv:
   - `project-a/` — a normal Python project.
   - `project-b/` — has its own package, e.g. installed via `pip install -e /some/external/source` into `project-b/.venv`, with the package's source living *outside* `project-b/` (so it can't be found via plain same-directory adjacency — this matters for a clean repro).
2. Open both as a multi-root workspace (e.g. via a `.code-workspace` file listing both folders).
3. Select a Python interpreter for the workspace (either venv).
4. Restart the Pylint server (`Pylint: Restart Server`).
5. Open a file in `project-b/` that imports its own package.

**Observed:** the file shows a false `import-error` for a package that is correctly installed in `project-b`'s own venv, because the language server is running under `project-a`'s interpreter (chosen because its path happens to be shorter).

## Evidence

Bootstrap root selection is purely based on shortest path (`vscode-common-python-lsp/typescript/src/utilities.ts`, `getProjectRoot()`):

```ts
for (const w of workspaces) {
    if (root && root.length > w.uri.fsPath.length && (await fs.pathExists(w.uri.fsPath))) {
        root = w.uri.fsPath;
        rootWorkspace = w;
    }
}
```

Running pylint directly, simulating exactly this scenario (bootstrap interpreter from the shorter-path folder, linting a file that lives in the other folder):

```
$ /path/to/project-a/.venv/bin/python -m pylint /path/to/project-b/main.py

************* Module main
/path/to/project-b/main.py:1:0: C0114: Missing module docstring (missing-module-docstring)
/path/to/project-b/main.py:1:0: E0401: Unable to import 'other_pkg' (import-error)

Your code has been rated at 0.00/10
```

The same file lints cleanly (no import-error) when run with its own venv's interpreter instead.

## Diagnostic Data
- Version of tool extension: current `main` (post `vscode-common-python-lsp` shared-package refactor, submodule commit `d1b98bc`)
- Pylint 4.0.5 / astroid 4.0.3 (bundled)
- Python 3.14.5, macOS

## Notes
Found while investigating #581 (which was a different, narrower bug about Global vs Workspace scope of `pylint.importStrategy` — that one appears already fixed, see the discussion there). This is a separate, still-open limitation in how the extension handles multiple independent Python environments within one multi-root workspace, since Pylint's `ToolConfig` doesn't opt into per-project environments (`supportsPerProjectEnvironments`) the way some other extensions built on the shared package might.

Possible directions:
- Opt Pylint into per-project environments / a language server instance per distinct interpreter, similar to whatever `supportsPerProjectEnvironments` already supports in the shared package.
- At minimum, surface a warning when a multi-root workspace has folders resolving to different interpreters, so users understand why non-bootstrap folders may show incorrect diagnostics.
